src/main/java/com/adamk33n3r/runelite/watchdog/EventHandler.java

```diff
@@ -189,26 +191,54 @@ private void onItemContainerChanged(ItemContainerChanged itemContainerChanged) {
         // Ignore everything but inventory
         if (itemContainerChanged.getItemContainer().getId() != InventoryID.INVENTORY.getId())
             return;
-        this.alertManager.getAllEnabledAlertsOfType(InventoryAlert.class)
-            .forEach(inventoryAlert -> {
-                Item[] items = itemContainerChanged.getItemContainer().getItems();
-                long itemCount = Arrays.stream(items).filter(item -> item.getId() > -1).count();
-                if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.FULL && itemCount == 28) {
-                    this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
-                } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.EMPTY && itemCount == 0) {
-                    this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
-                } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM) {
-                    Map<Integer, Integer> allItems = new HashMap<>();
-                    Arrays.stream(items)
-                        .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
-                    allItems.entrySet().stream()
-                        .filter(itemWithCount -> inventoryAlert.getItemQuantity() == 0 || itemWithCount.getValue() == inventoryAlert.getItemQuantity())
-                        .map(itemWithCount -> this.matchPattern(inventoryAlert, this.itemManager.getItemComposition(itemWithCount.getKey()).getName()))
-                        .filter(Objects::nonNull)
-                        .findFirst()
-                        .ifPresent(groups -> this.fireAlert(inventoryAlert, groups));
-                }
-            });
+        Item[] items = itemContainerChanged.getItemContainer().getItems();
+        long itemCount = Arrays.stream(items).filter(item -> item.getId() > -1).count();
+        Map<Integer, Integer> allItems = new HashMap<>();
+        Arrays.stream(items)
+                .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
+        // Skip firing alerts if there are no previous items, since we just logged in. Even an empty inventory will have a map of -1 itemIds.
+        if (!this.previousItemsTable.isEmpty()) {
```

Comment:
How will `isEmpty` ever be true? Or how will `!isEmpty` ever be true?

This is there to solve the same problem as the skill alerts, where they'd fire immediately upon login since they were being compared against a value that didn't exist yet. So this ensures that after login, the first inventory update (where isEmpty is true) will be skipped.

Reply:
Oh I think I realized what you are saying. Line 241 where this variable is assigned is _not_ inside of this if statement. It's right after it.

Reply:
Oh wow, my eyes must have been slantways hahahaha